src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -510,16 +510,26 @@ float FixedwingPositionControl::getCorrectedNpfgRollSetpoint()
 {
 	// Scale the npfg output to zero if npfg is not certain for correct output
 	float new_roll_setpoint(_npfg.getRollSetpoint());
-	const float can_run_factor(_npfg.canRun(_local_pos, _wind_valid));
+	const float can_run_factor(constrain(_npfg.canRun(_local_pos, _wind_valid), 0.f, 1.f));
 
-	if ((1.f - can_run_factor) < FLT_EPSILON) {
+	// Warn the user when the scale is less than 90% for at least 2 seconds.
+	if ((1.f - can_run_factor) < 0.1f) {
 		_need_report_npfg_uncertain_condition = true;
-	}
+		_time_since_first_reduced_roll = 0U;
+
+	} else if (_need_report_npfg_uncertain_condition) {
+		if (_time_since_first_reduced_roll == 0U) {
```

Comment:
We also need to reset _time_since_first_reduced_roll to 0 when we don't have NPFG running no?
We could store the last time getCorrectedNpfgRollSetpoint() is called and reset if that is above 2s. Or maybe you find a more elegant solution that comes without yet another class variable.

Reply:
Good catch, i will check how to best include that.